python/lsst/the/monster/isolate_and_transform.py

```diff
@@ -0,0 +1,170 @@
+import os
+from smatch import Matcher
+
+from lsst.pipe.tasks.isolatedStarAssociation import IsolatedStarAssociationTask
+import lsst.utils
+from .splinecolorterms import ColortermSpline
+from .refcats import GaiaXPInfo, GaiaDR3Info, SkyMapperInfo, PS1Info, VSTInfo
+from .utils import read_stars
+
+__all__ = ["MatchAndTransform"]
+
+"""
+This Python script takes a sharded catalog and outputs a shard transformed to
+DES bandpasses.
+
+The following steps are performed:
+1. Read in the Gaia shard.
+2. Run the IsolatedStarAssociationTask on the Gaia shard.
+3. Read in each catalog.
+4. Match each catalog with the Gaia DR3 catalog.
+5. For each band, interpolate the DES magnitudes.
+6. Save the catalog shard.
+"""
+
+
+class MatchAndTransform:
+    """Match catalogs to Gaia and transform them to 'the_monster'
+       reference frame.
+
+    Parameters
+    ----------
+    gaia_reference_class : `RefCatInfo`
+        The input Gaia DR3 RefcatInfo object.
+    catalog_info_list : `list` [`RefcatInfo`]
+        List of RefcatInfo objects for catalogs to transform.
+    write_path_inp : `str`, optional
+        The path to write the outputs to.
+    testing_mode : `bool`
+        Enter testing mode for read_stars?
+    """
+
+    def __init__(self,
+                 gaia_reference_class=GaiaDR3Info,
+                 catalog_info_list=[GaiaXPInfo, SkyMapperInfo, PS1Info, VSTInfo],
+                 write_path_inp=None,
+                 testing_mode=False,
+                 ):
+
+        self.gaia_reference_class = gaia_reference_class()
```

Comment:
And this should be `self.gaia_reference_info` on the left since it's not a class (and all the uses below).